lib/lightning/invocation.ex

```diff
@@ -270,11 +271,17 @@ defmodule Lightning.Invocation do
   """
   @spec get_run!(Ecto.UUID.t() | Event.t()) :: Run.t()
   def get_run!(%Event{id: event_id}) do
-    from(r in Run, where: r.event_id == ^event_id) |> Repo.one!()
+    from(r in Run, where: r.event_id == ^event_id)
+    |> Repo.one!()
   end
 
   def get_run!(id), do: Repo.get!(Run, id)
 
+  @doc """
+  Fetches a run and preloads the job via the run's event.
+  """
+  def get_run_with_job!(id), do: get_run!(id) |> Repo.preload(:job)
```

Comment:
I think we should either split out the query from `get_run!` into in it's own function, and then use it in both places.

We want to avoid `Repo.preload` as much as possible as it results in n+1 queries. 

So ideally the `_with_job!` variant is `from(r in run_query, preload: [:job]) |> Repo.one!()` etc.